Sources/TokamakCore/Views/List.swift

```diff
@@ -0,0 +1,398 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/2/20.
+//
+
+public struct List<SelectionValue, Content>: View
+  where SelectionValue: Hashable, Content: View {
+  public enum _Selection {
+    case one(Binding<SelectionValue?>?)
+    case many(Binding<Set<SelectionValue>>?)
+  }
+
+  var selection: _Selection
+  let content: Content
+
+  @Environment(\.listStyle) var style: ListStyle
+
+  public init(selection: Binding<Set<SelectionValue>>?, @ViewBuilder content: () -> Content) {
+    self.selection = .many(selection)
+    self.content = content()
+  }
+
+  public init(selection: Binding<SelectionValue?>?, @ViewBuilder content: () -> Content) {
+    self.selection = .one(selection)
+    self.content = content()
+  }
+
+  var listStack: some View {
+    VStack(alignment: .leading) { () -> AnyView in
+      if let contentContainer = content as? ParentView {
+        var sections = [AnyView]()
+        var currentSection = [AnyView]()
+        for child in contentContainer.children {
+          if child.view is SectionView {
+            if currentSection.count > 0 {
+              sections.append(AnyView(Section {
+                ForEach(Array(currentSection.enumerated()), id: \.offset) { _, view in
+                  view
+                }
+              }))
+              currentSection = []
+            }
+            sections.append(child)
+          } else {
+            if child.children.count > 0 {
+              currentSection.append(contentsOf: child.children)
+            } else {
+              currentSection.append(child)
+            }
+          }
+        }
+        if currentSection.count > 0 {
+          sections.append(AnyView(Section {
+            ForEach(Array(currentSection.enumerated()), id: \.offset) { _, view in
+              view
+            }
+          }))
+        }
+        return AnyView(_ListRow.buildItems(sections) { (view, isLast) -> AnyView in
+          if let section = view.view as? SectionView {
+            return AnyView(section.listRow(style)
+              .environment(\._outlineGroupStyle, _ListOutlineGroupStyle()))
+          } else {
+            return AnyView(_ListRow.listRow(view, style, isLast: isLast))
+          }
+        })
+      } else {
+        return AnyView(content)
+      }
+    }
+  }
+
+  public var body: some View {
+    if style is GroupedListStyle || style is InsetGroupedListStyle {
+      return AnyView(ScrollView {
+        HStack {
+          Spacer()
+        }
+        listStack
+      }
+      .padding(.top, 20)
+      .background(Color(0xEEEEEE))
```

Comment:
**suggestion:** Can we separate these out somewhere? Maybe they even should be renderer-specific?